wire-ios-request-strategy/Sources/Object Syncs/Helpers/IdentifierObjectSync.swift

```diff
@@ -106,20 +106,25 @@ public class IdentifierObjectSync<Transcoder: IdentifierObjectSyncTranscoder>: N
             switch response.result {
             case .permanentError, .success:
                 self.downloading.subtract(scheduled)
-                self.transcoder?.didReceive(response: response, for: scheduled)
-
-                if case .permanentError = response.result {
-                    self.delegate?.didFailToSyncAllObjects()
+                self.transcoder?.didReceive(response: response, for: scheduled) {
```

Comment:
Will we eventually be able to go another level higher with the asynchronous execution? 
So that `request.add(...)` receives something async and processes all the "completions" by gracefully waiting for all outstanding code to be done before moving to the next?

Reply:
I think that's difficult as long as we executing things inside the `OperationLoop` since it splits the request generating part from the processing of the response.

IMO the way forward is to execute things outside of the request loop, like the `MessageSender` does.